internal/handlers/common/aggregations/projection.go

```diff
@@ -0,0 +1,510 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggregations
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/iterator"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+)
+
+// errProjectionEmpty indicates projection field is empty.
+var errProjectionEmpty = errors.New("projection is empty")
+
+// ValidateProjection check projection document.
+// Document fields could be either included or excluded but not both.
+// Exception is for the _id field that could be included or excluded.
+//
+// Errors:
+//   - `ErrProjectionExIn` when there is exclusion in inclusion projection;
+//   - `ErrProjectionInEx` when there is inclusion in exclusion projection;
+//   - `ErrNotImplemented` when there is unimplemented projection operators and expressions;
+//   - `errProjectionEmpty` when projection document is empty.
```

Comment:
:warning: I don't think it is correct for the same exported function to return both protocol-error errors like `CommandError` with `ErrProjectionExIn` code and internal errors like `errProjectionEmpty`. Not to mention that documentation mixes protocol error codes and Go-level error values.